reVX/least_cost_xmission/least_cost_paths.py

```diff
@@ -380,3 +384,248 @@ def run(cls, cost_fpath, features_fpath, capacity_class,
                             (time.time() - ts) / 3600))
 
         return least_cost_paths
+
+
+class ReinforcementPaths(LeastCostPaths):
+    """
+    Compute reinforcement line paths between substations and a single
+    balancing area network node.
+    """
+    def __init__(self, cost_fpath, features, transmission_lines,
+                 xmission_config=None):
+        """
+        Parameters
+        ----------
+        cost_fpath : str
+            Path to h5 file with cost rasters and other required layers.
+        features_fpath : str
+            Path to GeoPackage with transmission features. The network
+            node must be the first row of the GeoPackage - the rest
+            should be substations that need to connect to that node.
+        transmission_lines :dict
+            Dictionary where the keys are the names of cost layers in
+            the cost HDF5 file and values are arrays with the
+            corresponding existing transmission lines rastered into
+            them (i.e. array value is 1 at a pixel if there is a
+            transmission line, otherwise 0). These arrays will be used
+            to compute the reinforcement costs along existing
+            transmission lines of differing voltages.
+        xmission_config : str | dict | XmissionConfig, optional
+            Path to Xmission config .json, dictionary of Xmission config
+            .jsons, or preloaded XmissionConfig objects.
+            By default, ``None``.
+        """
+        self._check_layers(cost_fpath)
+        self._config = TieLineCosts._parse_config(
+            xmission_config=xmission_config)
+
+        self._features, self._row_slice, self._col_slice, self._shape = \
+            self._map_to_costs(cost_fpath, features)
+        self._features = self._features.drop(columns='geometry')
+        self._cost_fpath = cost_fpath
+
+        network_node = (self._features.iloc[0:1]
+                        .dropna(axis="columns", how="all"))
+        err_msg = "Network node was dropped during clipping!"
+        assert "ba_str" in network_node, err_msg
+        self._start_idx = network_node[['row', 'col']].values[0]
```

Comment:
Nice so this is the central BA network node that the substations will be reinforced to, right? How do you feel about adding a property that references this for documentation purposes since this is so different than the _start_idx of the main LCP code? Actually it could be useful to have properties for both start and end indices for LCP and RP:

```
class LCP:
@property 
def start_indices():
"""List of starting index values that index the starting points of transmission features in the raster array"""
indices = self.features.index if indices is None else indices
indices = [self._features.loc[start, ['row', 'col']].values for start in indices]
return indices

def end_indices():
"""List of ending index values that index the termination points of transmission features in the raster array, this list corresponds to self.start_indices and is the same length"""
# do something similar but with that self._clip_to_feature() call


class RP:
@property 
def start_indices():
"""Starting index of the transmission feature, for RP this is the single central BA network node"""
return self._start_idx

@property 
def end_indices():
"""List of ending index values of the transmission features to connect the starting index to. For RP this is a list of substation indexes that need to be reinforced to the start index (central BA network node)"""
return self._features[['row', 'col']].values

Reply:
Ok, added those in! Let me know if you want to see any changes!